test/tests/syntax.json

```diff
@@ -697,6 +697,33 @@
     {
       "src": "{{trailing whitespace}} \n",
       "exp": "trailing whitespace"
+    },
+    {
+      "description": "NFC: literals are not normalized",
+      "src": "\u1E0A\u0323",
+      "exp": "\u1E0A\u0323"
+    },
+    {
+      "description": "NFC: variables are compared to each other as-if normalized; decl is non-normalized, use is",
+      "src": ".local $\u0044\u0323\u0307 = {foo} {{{$\u1E0c\u0307}}}",
+      "exp": "foo"
+    },
+    {
+      "description": "NFC: variables are compared to each other as-if normalized; decl is normalized, use isn't",
+      "src": ".local $\u1E0c\u0307 = {foo} {{{$\u0044\u0323\u0307}}}",
+      "exp": "foo"
+    },
+    {
+      "description": "NFC: variables are compared to each other as-if normalized; decl is normalized, use isn't",
+      "src": ".input {$\u1E0c\u0307} {{{$\u0044\u0323\u0307}}}",
+      "params": [{"name": "\u1E0c\u0307", "value": "foo"}],
+      "exp": "foo"
+    },
+    {
+      "description": "NFC: variables are compared to each other as-if normalized; decl is non-normalized, use is",
+      "src": ".input {$\u0044\u0323\u0307} {{{$\u1E0c\u0307}}}",
```

Comment:
You're missing the reordering case (`\u0044\u0307\u0323`)

You might include a special case mapping. See [Charmod example](https://www.w3.org/TR/charmod-norm/#example-15), featuring `U+01FA`==`U+00C5 U+0301`==`U+212B U+0301`==`U+0041 U+030A U+0301`. (212B is the Angstrom sign)

Reply:
Added these two tests in [80b73b7](https://github.com/unicode-org/message-format-wg/pull/904/commits/80b73b71dd30809709737a847bc9bea8037713b8)